subsys/bluetooth/host/hci_core.c

```diff
@@ -4276,7 +4276,10 @@ int bt_le_set_chan_map(uint8_t chan_map[5])
 	struct bt_hci_cp_le_set_host_chan_classif *cp;
 	struct net_buf *buf;
 
-	if (!IS_ENABLED(CONFIG_BT_CENTRAL)) {
+	if (!(IS_ENABLED(CONFIG_BT_CENTRAL)
+	    | IS_ENABLED(CONFIG_BT_ISO_BROADCASTER)
+	    | IS_ENABLED(CONFIG_BT_ISO_CENTRAL)
+	    | IS_ENABLED(CONFIG_BT_PER_ADV))) {
```

Comment:
I wonder if this could be simplified as `IS_ENABLED(CONFIG_BT_CENTRAL) || IS_ENABLED(CONFIG_BT_BROADCASTER)` as all the higher layer roles (such as `CONFIG_BT_ISO_BROADCASTER` and `CONFIG_BT_ISO_CENTRAL`) will depend on either of those.